modules/fedimint-ln-server/src/lib.rs

```diff
@@ -377,167 +377,157 @@ impl ServerModule for Lightning {
+                    FundedContract::Outgoing(..) => {
+                        bail!("Contract account for this decryption share is outgoing");
                     }
+                };
 
-                    let account = match self.get_contract_account(dbtx, contract_id).await {
-                        Some(account) => account,
-                        None => {
-                            warn!("Received decryption share for non-existent contract account");
-                            continue;
-                        }
-                    };
+                if contract.decrypted_preimage != DecryptedPreimage::Pending {
+                    bail!("Contract for this decryption share is not pending");
+                }
 
-                    let (contract, out_point) = match account.contract {
-                        FundedContract::Incoming(contract) => {
-                            (contract.contract, contract.out_point)
-                        }
-                        FundedContract::Outgoing(..) => {
-                            warn!("Received decryption share for outgoing contract");
-                            continue;
-                        }
-                    };
+                if !self.validate_decryption_share(peer_id, &share, &contract.encrypted_preimage) {
+                    bail!("Decryption share is invalid");
+                }
 
-                    if contract.decrypted_preimage != DecryptedPreimage::Pending {
-                        warn!("Received decryption share for non-pending incoming contract");
-                        continue;
-                    }
+                // we save the first ordered valid decryption share for every peer
+                dbtx.insert_new_entry(&AgreedDecryptionShareKey(contract_id, peer_id), &share)
+                    .await;
 
-                    if !self.validate_decryption_share(
-                        peer_id,
-                        &share,
-                        &contract.encrypted_preimage,
-                    ) {
-                        warn!("Received invalid decryption share for incoming contract");
-                        continue;
-                    }
+                // collect all valid decryption shares previously received for this contract
+                let decryption_shares = dbtx
+                    .find_by_prefix(&AgreedDecryptionShareContractIdPrefix(contract_id))
+                    .await
+                    .map(|(key, decryption_share)| (key.1, decryption_share))
+                    .collect::<Vec<_>>()
+                    .await;
 
-                    // we save the first ordered valid decryption share for every peer
-                    dbtx.insert_new_entry(&AgreedDecryptionShareKey(contract_id, peer_id), &share)
-                        .await;
+                if decryption_shares.len() < self.cfg.consensus.threshold() {
```

Comment:
There appear to be some changes in here that are hard to review since they are mixed with reformatting. I think we can go ahead, but maybe these should have been separate commits doing these changes before the for loop unroll.